### ISSUE TITLE
Users can add works by other users to their own works

### ISSUE THREAD

**Issue by julesies** [projecthydra/sufia#3042]
Users can add any work (child resource?), including the work of others, to their own work and can add their work as a child work to parents, not their own. 

We have a Sufia 7.2 application we are testing. We found this behavior surprising and wanted to see if it was a bug. 

Expected behavior:
Regular/non-admin users are not able to add other people works as a child resource to their own unless they have explicitly been given permission to do so by the owner of the work. Admin users would be able to add works to work by different depositors if needed. 

Actual behavior:
Regular non-admin user can add any work to their own work - results are that one can add works they didn't author to their own works. There is no public facing indication that the work is not their own. You can add your own work as a child of another parent work not owned by you, as well. 

1. As a non-admin user, create a work
2. In the Relationships tab under "Other Works in this Work", add the id of a work created by another user.

**Comment by mjgiarlo** [comment]
@julesies Hi, Julie. :smile: 

Yes, I believe this was resolved with the following PR that will be in 7.3.0: https://github.com/projecthydra/sufia/pull/3015 

Might y'all have cycles to test this again with a more recent build of Sufia to confirm?

Thank you for reporting this!

**Comment by julesies** [comment]
@mjgiarlo yes we can try this. Is this why users can add their works to admin sets and collection that they do not own? It's still a little challenging for us to tell how things are supposed to work vs bug vs local problem. Thanks for your help 😸

**Comment by mjgiarlo** [comment]
@julesies I'm not sure we've worked on -- or were even aware of -- the admin set/collection issues you raised. Do you mind writing up another issue about those?

Happy to help. Thx!

**Comment by julesies** [comment]
@mjgiarlo sorry, adding to user collections is working fine. I wrote up the Admin Set behavior that is confusing.

**Comment by julesies** [comment]
@mjgiarlo We upgraded to 7.3.0rc1 and this behavior seems the same. However, now when I hit save I see a message "another user has made a change to that Work since you accessed the edit form". I don't believe that another user was working on that item. Then I click save again and the change I made (adding another person's work to mine) saves.

**Comment by mjgiarlo** [comment]
@julesies Thanks for looking into this! I've made this a blocking bug for the 7.3.0 release.

**Comment by vantuyls** [comment]
So, I think it would be useful for us to get a little community clarity on what the expected behavior is. I'm not convinced that a user should be able to insert another user's object into their PCDM model (though i'm willing to be convinced). The relationship i could see making more sense would be to link to the existing object. What is the use case for one user adding content to another user's deposit?

@mjgiarlo @julesies @scholarworks  (also, i cant seem to tag Hannah Frost on this)

**Comment by vantuyls** [comment]
i misread the original issue (had it flipped around). ignore my above comment. thanks for identifying and working on this, all.

**Comment by julesies** [comment]
@vantuyls Yeah I did not make it super clear in the title. I'm glad to hear more agreement on the behavior.

**Comment by hortongn** [comment]
Now that https://github.com/projecthydra/curation_concerns/pull/1155 has been merged, I can't reporduce this bug anymore.  Do we consider this fixed or is there more work to do?

**Comment by mjgiarlo** [comment]
@hortongn (Did Sufia already pick up that change in CC, or did you tweak your app Gemfile to get it?)

I believe the remaining work is under way by @revgum to use ability checks rather than depositor comparison, and once that work is merged into `master` and `1-7-stable`, I'd like to have a new 1.7.x release cut, at which point we should test with vanilla sufia master. (If all's good then, yes, let's close this issue and consider cutting an rc2?) Cc: @jcoyne

**Comment by hortongn** [comment]
@mjgiarlo I had Sufia using the CC 1-7-stable branch.  Was trying to get a sense of how @revgum's PR affected this bug.

Good to know that a tweak to use ability checks is in the works.  Thanks for the update!

**Comment by mjgiarlo** [comment]
This should be fixed by https://github.com/projecthydra/sufia/pull/3118.
